### ISSUE TITLE
consumerCount on disabled connection

### ISSUE THREAD

**Issue by vkfont** [dropbox/amqp-coffee#28]
if i will call queue.consumerCount or queue.messageCount durning connection problems - it stops fire callbacks, even if connection will back later, is something remain closed?

idea to fix:
- set flag to true on amqp ready event, and set to false on amqp close event
- return specific error in consumerCount/messageCount callback if flag is not active

**Comment by barshow** [comment]
I can't repro this, I did do a small race fix that may help.  Its also in https://github.com/dropbox/amqp-coffee/pull/27

If you have repro steps or turn on debugging `AMQP=3 DEBUG=*` and send some logs ill have look.

**Comment by vkfont** [comment]
1) to stop receiving consumerCount after reconnect, try this:
- launch code https://gist.github.com/vkfont/65d77cf461c075ebba60
- restart service
- if you get similar output (`consumers count: 1` not displaying after reconnect) then bug is reproducing:
```
connected to rabbit@Andreys-Air v3.5.0
queue "named-queue-test" created
message received: #1
try to get consumers count for queue...
consumers count: 1
message received: #2
try to get consumers count for queue...
consumers count: 1
message received: #3
try to get consumers count for queue...
consumers count: 1
message received: #4
connection is closed: 3634ms
try to get consumers count for queue...
try to get consumers count for queue...
try to get consumers count for queue...
try to get consumers count for queue...
connected to rabbit@Andreys-Air v3.5.0
message received: #5
try to get consumers count for queue...
message received: #6
try to get consumers count for queue...
message received: #7
try to get consumers count for queue...
message received: #8
try to get consumers count for queue...
message received: #9
```

2) i get error during .consumerCount call  on unnamed queue, but im not sure is it bug of protocol feature:
```
/Users/drago/Documents/amqp-coffee/bin/src/lib/Queue.js:133
        return cb(err, res.consumerCount);
                          ^
TypeError: Cannot read property 'consumerCount' of undefined
    at /Users/drago/Documents/amqp-coffee/bin/src/lib/Queue.js:133:27
    at /Users/drago/Documents/amqp-coffee/bin/src/lib/Queue.js:69:43
    at doneFn (/Users/drago/Documents/amqp-coffee/bin/src/lib/Channel.js:241:11)
    at TemporaryChannel.Channel._onChannelMethod (/Users/drago/Documents/amqp-coffee/bin/src/lib/Channel.js:336:71)
    at Connection._onMethod (/Users/drago/Documents/amqp-coffee/bin/src/lib/Connection.js:601:39)
    at AMQPParser.<anonymous> (/Users/drago/Documents/amqp-coffee/bin/src/lib/Connection.js:4:61)
    at AMQPParser.emit (events.js:118:17)
    at AMQPParser.parseMethodFrame (/Users/drago/Documents/amqp-coffee/bin/src/lib/AMQPParser.js:110:19)
    at AMQPParser.frameEnd (/Users/drago/Documents/amqp-coffee/bin/src/lib/AMQPParser.js:80:16)
    at AMQPParser.frame (/Users/drago/Documents/amqp-coffee/bin/src/lib/AMQPParser.js:63:21)
```

**Comment by barshow** [comment]
Thanks for that repro!  Its fixed in https://github.com/dropbox/amqp-coffee/pull/27 now and wrote some tests based on your repro.

**Comment by barshow** [comment]
its published to npm as 0.1.21-rc1 right now if you want to try it

**Comment by vkfont** [comment]
cook, thank you
everything works
